File: 7_kyu/Alphabetically_ordered.c
#include <stdbool.h>
#include <stddef.h>

// Checks if the characters in the string are ordered.
// Returns true if all string[i] <= string[j] with i <= j.
bool is_ordered (const char string[] )
{
  if( string == NULL || string[0] == '\0' )
    return true;
  // Check the char pairs
  for( register int i = 1; string[i]; ++i )
    if( string[i-1] > string[i] )
      return false;
  // All chars are orders
	return true;
}

File: 7_kyu/Alphabetically_ordered.rs
// Checks if characters in string s are sorted.
fn alphabetic(s: &str) -> bool {
    let mut chars = s.chars();
    // Set last_char to '0' if string is empty.
    // It's ok because the char will be never used.
    let mut last_char: char = chars.next().unwrap_or('0');
    // Attention: the loop starts at the second char of s.
    for c in chars {
        if c < last_char {
            // Quick exit if string is unsorted
            return false;
        }
        last_char = c;
    }
    // All chars in the string s are checked.
    // The string is sorted.
    true
}
